### PR TITLE
CAW-6: Add dependency-cruiser dev tool

### PR DESCRIPTION
**Pull request by bvarberg**
## Description for [CAW-6](https://crema.atlassian.net/browse/CAW-6)

Adds the `dependency-cruiser` package for managing, validating, and visualizing dependencies (i.e. imports).

> **Note:** This PR includes the  "recommended" rules from `dependency-cruiser` and does not add any custom rules.

### 🖼 Demo Images

Screen Shot 2020-01-10 at 11 38 10 AM
> Above: `test:deps` task, for CI

Screen Shot 2020-01-10 at 11 39 27 AM
> Above: `deps:graph` task, for local visualization

Screen Shot 2020-01-10 at 11 40 31 AM
> Above: `deps:report` task, for a more human-friendly validation report (it explains the rules), potentially used for CI output reports

### 👩‍🔬 Test Instructions

1. Run `npm install` to install the new dependencies
2. Run `npm run test:deps` to validate dependencies in the `src` directory

## 🔎 Reviewer Checklist

> Checked off by the PR **Reviewers**

### Required

> These always need to be checked

- [x] Merge destination is correct
- [x] Code is correct as understood and conforms to quality standards
- [x] Tests have been added where appropriate (unit, visual, end-to-end)
- [x] Acceptance Criteria have been met

---

>### Roles & Responsibilities
>
>#### 👨‍💻 Assignee
>
>- Initiator of this PR (be sure to set in GitHub UI)
>- Addresses feedback and change requests
>- Merges PR once approved (usually deletes branch unless `develop` or `release`)
>
>#### 👩‍💻 Reviewer
>
>- Invited to review PR by **Assignee** (via GitHub UI)
>- Is expected to complete a review and address followup

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,7 +49,10 @@
     "new:util": "hygen util new",
     "prepush": "npm run test:ci",
     "static": "serve -s build",
+    "deps:graph": "depcruise --validate --output-type dot -- src | dot -T svg > dependency-graph.svg",
+    "deps:report": "depcruise --validate --output-type err-html --output-to dependency-report.html -- src",
     "test:ci": "CI=true npm-run-all test:lint test:types \"test:unit -- -i\" test:visual:ci test:e2e:ci",
```
Comment (roblafeve) [inline package.json]: Maybe we should add `test:deps` to `test:ci`?

**Review by roblafeve — COMMENTED** [review]
This is looking great, I'm going to play around with it a bit more to get a feel for it.

File: package.json [hunk 2]
```diff
@@ -49,7 +49,10 @@
     "new:util": "hygen util new",
     "prepush": "npm run test:ci",
     "static": "serve -s build",
+    "deps:graph": "depcruise --validate --output-type dot -- src | dot -T svg > dependency-graph.svg",
```
Comment (roblafeve) [inline package.json:52]: When I run this script I get this error:
```
▶ npm run deps:graph

> crema-app-web@0.1.0 deps:graph /Users/Rob_LaFeve/Dev/crema-app-web
> depcruise --validate --output-type dot -- src | dot -T svg > dependency-graph.svg

sh: dot: command not found

  ERROR: write EPIPE
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! crema-app-web@0.1.0 deps:graph: `depcruise --validate --output-type dot -- src | dot -T svg > dependency-graph.svg`
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the crema-app-web@0.1.0 deps:graph script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/Rob_LaFeve/.npm/_logs/2020-01-10T18_37_40_936Z-debug.log
```

An `svg` is produced but it is blank.

Comment (roblafeve) [reply package.json:52]: Looks like a Graphviz issue?

File: package.json (hunk 1 above)
Comment (bvarberg) [reply package.json]: Ah good idea.

File: package.json (hunk 2 above)
Comment (bvarberg) [reply package.json:52]: @roblafeve Did you have Graphviz installed at the time of running this?

Comment (roblafeve) [reply package.json:52]: I did not. I ran `brew install graphviz` and then it worked as expected. I simply missed your note on the script ;) I think this is good 👍

File: package.json (hunk 1 above)
Comment (roblafeve) [reply package.json]: I would recommend adding it toward the beginning of these (maybe after lint?) so that it fails fast 👍

Comment (bvarberg) [reply package.json]: Yep, I had added it after the types -- cool with that positioning?
4482a8e

**Comment by blaketarter** [comment]
@bvarberg Would we want this to run in the Github Actions CI? If so you may want to look into adding it somewhere around [here](https://github.com/cremalab/crema-app-web/blob/develop/.github/workflows/CI.yml#L20).

**Comment by roblafeve** [comment]
I'm totally cool with adding the custom script that we've been discussing ;)

**Comment by Alexandria** [comment]
Just ran this myself and it works just fine for me👍 Does running `npm run test:deps` also generate the graph? If that is the case, I did not get a graph.

**Comment by roblafeve** [comment]
@Alexandria it doesn't currently—you'd have to install `graphviz` on your system for the script `npm run deps:graph` to generate the svg. I did a `brew install graphviz` and I was good to go.

**Review by roblafeve — APPROVED** [review]
I think this is great 👍

(still pro-adding the one-level dependency rule BTW 😄 )

**Pull request merged**